www/ebooks/download.php

```diff
@@ -40,11 +45,15 @@
 	/** @var string|null $userAgent */
 	$userAgent = $_SERVER['HTTP_USER_AGENT'] ?? null;
 
-	try{
-		$ebook->AddDownload($ipAddress, $userAgent);
+	// Check for excessive downloads.
```

Comment:
If the user is logged in,  or if we're from a `source`, then we can skip rate limiting downloads, on the assumption that patrons are going to be polite.

Reply:
I added a check if the user is logged in, but since all file downloads come from `source=feed` or `source=download`, I didn't add a check for `source`. (I also thought it would be each for someone writing a script to add.)